distributionviewer/core/static/js/app/components/containers/chart-detail-container.js

```diff
@@ -16,20 +17,25 @@ class ChartDetailContainer extends React.Component {
   }
 
   render() {
-    return (
-      <ChartDetail
-        id={this.chartId}
-        metric={this.props.metric.metric}
-        points={this.props.metric.points}
-        type={this.props.metric.type}
-      />
-    );
+    if (this.props.status === 404) {
+      return <NotFound />;
+    } else {
+      return (
+        <ChartDetail
+          id={this.chartId}
+          metric={this.props.metric.metric}
+          points={this.props.metric.points}
+          type={this.props.metric.type}
+        />
+      );
+    }
   }
 }
 
 const mapStateToProps = function(store) {
   return {
-    metric: store.metricState.metric
+    metric: store.metricState.metric,
+    status: store.metricState.status,
```

Comment:
And you're right, it looks like Babel does [remove trialing commas](https://babeljs.io/repl/#?babili=false&evaluate=true&lineWrap=false&presets=es2015%2Creact%2Cstage-2&code=var%20obj%20%3D%20%7B%0A%20%20a%3A%201%2C%0A%20%20b%3A%202%2C%0A%7D) just to play it safe.